src/Jackalope/ObjectManager.php

```diff
@@ -818,12 +818,12 @@ public function save()
             $this->executeOperations($this->operationsLog);
 
             // loop through cached nodes and commit all dirty and set them to clean.
-            if (isset($this->objectsByPath['Node'])) {
-                foreach ($this->objectsByPath['Node'] as $node) {
+            foreach ($this->objectsByPath as $nodes) {
```

Comment:
maybe just something like `if ($this->transport instanceof ...) {$this->saveVersionNodes();}` where you implement that loop. or move the loop into a method with the class as parameter, to avoid duplication. then always call it with Node and possibly with Version and so on.